flows/definition/flow.go

```diff
@@ -32,18 +32,18 @@ type flow struct {
 	// optional properties not used by engine itself
 	ui json.RawMessage
 
-	// properties set after validation
+	// properties set after inspection
```

Comment:
feeling less sure about this approach of adding unofficial fields to the flow definition, especially now that in RP we save these off to Flow.metadata.. maybe the validate endpoint should just return this stuff separately

Reply:
U-Report will use this information but also not from the definition directly - instead we'll expose parts of `Flow.metadata` on the flows.json endpoint

Reply:
Other question may be what this means from an import / export perspective. IE, do we include metadata in those exports? (I think there was some thought at some point that the exports may contain additional information such as the type of contact fields we depend on so that we can recreate them appropriately, but that may be separate)

Reply:
Hmm contact field types is an interesting one - thus far everything we've called metadata is stuff that we can re-calculate from the flow definition - so there's no reason to trust incoming metadata when the goflow validate call is going to return refreshed metadata. If we did add something like contact field types then it would be more analogous to `_ui` in that the engine doesn't want/need it but RapidPro does.

So I think that's maybe (?) an argument for treating these re-calculable things ( `_results`/`_waiting_exits`/`_dependencies` as something different which are not kept in the flow definition and are not included in exports. They're just a thing you get back from mailroom/validate and RP can put them wherever it wants. 

I guess the thing that's irking me most is having a growing number of top-level "unofficial" parts of the spec with no clear demarcation of who owns and needs what. `_ui` is nice because the editor owns that and can work with any flow that has just that.

Reply:
Since this PR doesn't change that stuff, am punting this conversation to a new issue #679